Compiler warnings in RigidBodyTree when using GCC

Four warnings along the lines of this one: 

```
multibody/rigid_body_tree.cc:110:63: warning: 'RigidBodyTree::bodies' is deprecated: 
DRAKE DEPRECATED: Direct access to `bodies` has been deprecated, mutable access has been removed. Use `get_bodies` and `add_rigid_body` instead [-Wdeprecated-declarations]
     : collision_model_(drake::multibody::collision::newModel()) {
                                                               ^
In file included from multibody/rigid_body_tree.cc:1:0:
bazel-out/k8-dbg/bin/multibody/_virtual_includes/rigid_body_tree_cc/drake/multibody/rigid_body_tree.h:1526:67: note: declared here
   const std::vector>>& bodies{bodies_};
```

--- a/multibody/rigid_body_tree.cc
+++ b/multibody/rigid_body_tree.cc
@@ -105,9 +105,13 @@ std::ostream& operator<<(std::ostream& os, const RigidBodyTree<double>& tree) {
   return os;
 }
 
+// Suppress deprecation warnings when constructing `bodies` and `frames`.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 template <typename T>
 RigidBodyTree<T>::RigidBodyTree()
     : collision_model_(drake::multibody::collision::newModel()) {
+#pragma GCC diagnostic pop  // pop -Wdeprecated-declarations
   // Sets the gravity vector.
   a_grav << 0, 0, 0, 0, 0, -9.81;
 

